deps: update to github.com/lestrrat-go/jwx/v2

kolyshkin opened the pull request:
Version 1 is using some outdated packages (like pkg/errors) and is generally obsolete.

Using an older v2 version as newer ones bump go in go.mod which I'm
trying to avoid here. Same issue with v3.

Closes: #18

kolyshkin committed:
deps: update to github.com/lestrrat-go/jwx/v2

Version 1 is using some outdated packages (like pkg/errors) and is
generally obsolete.

Using an older v2 version as newer ones bump go in go.mod which I'm
trying to avoid here. Same issue with v3.

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

anmaxvl commented:
@KenGordon any concerns with merging this?

kolyshkin commented:
@KenGordon PTAL

gibmat commented:
And `github.com/lestrrat-go/jwx/v3` is now being used; the relevant change is quite simple once again:

```
diff --git a/pkg/did-x509-resolver/resolver.go b/pkg/did-x509-resolver/resolver.go
index 8eb373c..ee3576b 100644
--- a/pkg/did-x509-resolver/resolver.go
+++ b/pkg/did-x509-resolver/resolver.go
@@ -14,7 +14,7 @@ import (
        "strconv"
        "strings"
 
-       "github.com/lestrrat-go/jwx/jwk"
+       "github.com/lestrrat-go/jwx/v3/jwk"
 )
 
 func VerifyCertificateChain(chain []*x509.Certificate, trustedRoots []*x509.Certificate, ignoreTime bool) ([][]*x509.Certificate, error) {
@@ -392,7 +392,7 @@ func createDidDocument(did string, chain []*x509.Certificate) (string, error) {
                ka = fmt.Sprintf(",\"keyAgreement\": \"%s#key-1\"", did)
        }
 
-       leaf, err := jwk.New(chain[0].PublicKey)
+       leaf, err := jwk.Import(chain[0].PublicKey)
        if err != nil {
                return "", err
        }
```

Bumping jwx to v3 and tagging a new release would be helpful for Debian's packaging of this library.

kolyshkin commented:
@gibmat not quite -- apparently github.com/lestrrat-go/jwx/v3 requires go 1.24, while this repo has go 1.20 in its go.mod.

Besides, it seems that this repo is hopelessly abandoned, as this PR is 1+ year old and no one cared to even comment on this.

Having said that, I've updated to the latest v2 release which do not require changing go version in go.mod, and updated the commit message accordingly (as well as the PR description).

kolyshkin commented:
@MahatiC @takuro-sato PTAL 🙏🏻

KenGordon commented:
> @gibmat not quite -- apparently github.com/lestrrat-go/jwx/v3 requires go 1.24, while this repo has go 1.20 in its go.mod.
> 
> Besides, it seems that this repo is hopelessly abandoned, as this PR is 1+ year old and no one cared to even comment on this.
> 
> Having said that, I've updated to the latest v2 release which do not require changing go version in go.mod, and updated the commit message accordingly (as well as the PR description).

What is your interest here?

kolyshkin commented:
> What is your interest here?

This is pretty simple. I was assessing dependencies of a package (don't remember now which one exactly) for using obsoleted package `github.com/pkg/errors`, and as a result opened a few PRs to fix this, with the ultimate goal to get rid of github.com/pkg/errors (indirect) dependency in the above package.
